core/templates/pages/blog-dashboard-page/blog-dashboard-tile/blog-dashboard-tile.component.spec.ts

```diff
@@ -97,4 +126,104 @@ describe('Blog Dashboard Tile Component', () => {
       expect(component.getDateStringInWords(DATE))
         .toBe('February 2, 2018');
     });
+
+  it('should navigate to blog post editor interface on clicking edit', () => {
+    spyOn(blogDashboardPageService, 'navigateToEditorTabWithId');
+    component.blogPostSummary = BlogPostSummary.createFromBackendDict(
+      sampleBlogPostSummary);
+
+    component.editBlogPost();
+
+    expect(blogDashboardPageService.navigateToEditorTabWithId)
+      .toHaveBeenCalledWith('sampleId');
+  });
+
+  it('should successfully place call to delete blog post model', fakeAsync(
+    () => {
+      spyOn(ngbModal, 'open').and.returnValue({
+        result: Promise.resolve()
+      } as NgbModalRef);
+      spyOn(blogDashboardPageService, 'deleteBlogPost');
+      component.blogPostSummary = BlogPostSummary.createFromBackendDict(
+        sampleBlogPostSummary);
+
+      component.deleteBlogPost();
+      tick();
```

Comment:
Why do we need `tick()` here?

Reply:
![Screenshot from 2021-08-14 01-36-47](https://user-images.githubusercontent.com/68547101/129413101-8865bbf7-efbb-4541-9b73-b1ee16f8c2e3.png)
Because we are resolving a promise here.

Reply:
I don't think we need `tick()` to resolve the promise returned. Can you check other files where `spyOn(ngbModal, 'open')` is used?

Reply:
It does not work without it,  I just removed the tick() statement:
![Screenshot from 2021-08-15 19-03-37](https://user-images.githubusercontent.com/68547101/129480439-1fdf9407-6d78-4192-8642-f54822cd26c3.png)

Reply:
ah, thanks for the explanation